js/pages/data-sources/components/reports/treemapDrilldown.js

```diff
@@ -282,26 +305,27 @@ define([
 
 		getData() {
 			const response = super.getData();
-			// immediately hide report loader
-			this.context.loadingReport(false);
-
 			return response;
 		}
 
 		loadData(selectedConcept) {
 			this.conceptId = selectedConcept.concept_id;
 			this.currentConcept(selectedConcept);
 			this.isError(false);
-			this.isLoading(true);
 			this.getData()
-				.then((data) => this.parseData(data))
+				.then((data) => {
+					this.parseData(data);
+					this.context.model.loadingReportDrilldown(false);
+					this.scrollTo("#datasourceReportDrilldownTitle");
+				})
 				.catch((er) => {
+					this.context.model.loadingReportDrilldown(false);
```

Comment:
those two lines could be moved to `finally` block

Reply:
resolved in new commit commit.